core/src/read/read_stream.rs

```diff
@@ -756,58 +756,84 @@ impl<D: Decoder> ReadDiskStream<D> {
             reached_end_of_file = true;
         }
 
+        let copy_block_into_read_buffer =
+            |heap: &mut HeapData<<D as Decoder>::T>,
+             block_index: usize,
+             start_frame_in_block: usize,
+             frames: usize| {
+                let maybe_block = {
```

Comment:
Using let-else statements with early returns might improve the readability.

Reply:
Oh, I didn't know Rust had let-else statements. Awesome, learn something new every day.